### PR TITLE
[MONIT-25838] Consolidate Proxy pom.xml

### PR DESCRIPTION
**Pull request by laullon**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,22 +5,25 @@
...
 ## To start developing
 
-```
-$ git clone https://github.com/wavefronthq/wavefront-proxy
-$ cd wavefront-proxy
-$ mvn clean install -DskipTests
+```bash
+git clone https://github.com/wavefronthq/wavefront-proxy
+cd wavefront-proxy
+vn -f proxy clean install -DskipTests
```
Comment (joannak-vmware) [inline README.md:24]: should this be `mvn`, not `vn`?

File: mvnw [hunk 2]
```diff
@@ -0,0 +1,2 @@
+set -x
+mvn -f proxy $@
```
Comment (joannak-vmware) [inline mvnw:2]: general question, what is this file for? I don't see it used in other file here.

File: proxy/src/main/java/com/wavefront/agent/AbstractAgent.java [hunk 3]
```diff
@@ -187,8 +187,9 @@ private void postProcessConfig() {
     // connection. the alternative is to always validate connections before reuse, but since
     // it happens fairly infrequently, and connection re-validation performance penalty is
     // incurred every time, suppressing that message seems to be a more reasonable approach.
-    org.apache.log4j.Logger.getLogger("org.apache.http.impl.execchain.RetryExec").
-        setLevel(org.apache.log4j.Level.WARN);
+    // org.apache.log4j.Logger.getLogger("org.apache.http.impl.execchain.RetryExec").
+    //     setLevel(org.apache.log4j.Level.WARN);
+    Logger.getLogger("org.apache.http.impl.execchain.RetryExec").setLevel(Level.WARNING);
```
Comment (joannak-vmware) [inline proxy/src/main/java/com/wavefront/agent/AbstractAgent.java:192]: I don't see a level `WARNING` level in [log4j docs,](https://logging.apache.org/log4j/1.2/apidocs/org/apache/log4j/Level.html) did you mean `WARN`?

**Review by joannak-vmware — COMMENTED** [review]
Added some comments to address before approving PR. Thanks!

**Comment by joannak-vmware** [comment]
Could we also bump `wavefront-internal-reporter` version to 1.7.10 to resolve slf4j vulnerability?

File: proxy/src/main/java/com/wavefront/agent/AbstractAgent.java (hunk 3 above)
Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/AbstractAgent.java:192]: is java.util.logging... but i need to review if that still works, and if is need it

File: mvnw (hunk 2 above)
Comment (laullon) [reply mvnw:2]: is for travisCI, but will remove it, doesn't make sense

**Comment by laullon** [comment]
> Could we also bump `wavefront-internal-reporter` version to 1.7.10 to resolve slf4j vulnerability? Edit: Had overlooked, thuis fix is already in proxy/pom.xml

Is done, the SDK 3.0 already have wavefront-internal-reporter  1.7.10 as dependency, so no need to spiffy it here.

```
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ proxy ---
[INFO] com.wavefront:proxy:jar:11.0-RC3-SNAPSHOT
[INFO] +- com.wavefront:java-lib:jar:2020-12.1:compile
[INFO] |  +- com.google.guava:guava:jar:31.0.1-jre:compile
[INFO] |  |  +- com.google.guava:failureaccess:jar:1.0.1:compile
[INFO] |  |  +- com.google.guava:listenablefuture:jar:9999.0-empty-to-avoid-conflict-with-guava:compile
[INFO] |  |  \- com.google.j2objc:j2objc-annotations:jar:1.3:compile
[INFO] |  +- javax.validation:validation-api:jar:1.1.0.Final:compile
[INFO] |  +- joda-time:joda-time:jar:2.10.12:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-databind:jar:2.12.3:compile
[INFO] |  +- org.jboss.resteasy:resteasy-jaxrs:jar:3.15.2.Final:compile
[INFO] |  |  +- org.jboss.spec.javax.xml.bind:jboss-jaxb-api_2.3_spec:jar:2.0.1.Final:compile
[INFO] |  |  +- org.reactivestreams:reactive-streams:jar:1.0.3:compile
[INFO] |  |  +- jakarta.validation:jakarta.validation-api:jar:2.0.2:compile
[INFO] |  |  +- org.jboss.spec.javax.annotation:jboss-annotations-api_1.3_spec:jar:2.0.1.Final:compile
[INFO] |  |  +- com.sun.activation:jakarta.activation:jar:1.2.2:compile
[INFO] |  |  \- com.github.stephenc.jcip:jcip-annotations:jar:1.0-1:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-core:jar:2.12.3:compile
[INFO] |  +- org.apache.avro:avro:jar:1.9.2:compile
[INFO] |  |  \- org.apache.commons:commons-compress:jar:1.21:compile
[INFO] |  +- io.netty:netty-handler:jar:4.1.70.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:linux-x86_64:4.1.70.Final:compile
[INFO] |  |  +- io.netty:netty-transport-native-unix-common:jar:4.1.70.Final:compile
[INFO] |  |  \- io.netty:netty-transport-classes-epoll:jar:4.1.70.Final:compile
[INFO] |  +- net.razorvine:pyrolite:jar:4.10:compile
[INFO] |  |  \- net.razorvine:serpent:jar:1.12:compile
[INFO] |  +- io.dropwizard.metrics5:metrics-core:jar:5.0.0-rc2:compile
[INFO] |  +- dk.brics:automaton:jar:1.12-1:compile
[INFO] |  +- com.google.re2j:re2j:jar:1.2:compile
[INFO] |  \- org.antlr:antlr4-runtime:jar:4.7.2:compile
[INFO] +- org.jboss.resteasy:resteasy-bom:pom:3.13.0.Final:compile
[INFO] +- com.fasterxml.jackson.module:jackson-module-afterburner:jar:2.12.3:compile
[INFO] +- com.github.ben-manes.caffeine:caffeine:jar:2.8.0:compile
[INFO] |  +- org.checkerframework:checker-qual:jar:3.18.1:compile
[INFO] |  \- com.google.errorprone:error_prone_annotations:jar:2.9.0:compile
[INFO] +- org.apache.httpcomponents:httpclient:jar:4.5.13:compile
[INFO] |  +- org.apache.httpcomponents:httpcore:jar:4.4.13:compile
[INFO] |  +- commons-logging:commons-logging:jar:1.2:compile
[INFO] |  \- commons-codec:commons-codec:jar:1.15:compile
[INFO] +- org.apache.logging.log4j:log4j-api:jar:2.14.1:compile
[INFO] +- com.yammer.metrics:metrics-core:jar:2.2.0:compile
[INFO] |  \- org.slf4j:slf4j-api:jar:1.8.0-beta4:compile
[INFO] +- commons-lang:commons-lang:jar:2.6:compile
[INFO] +- commons-collections:commons-collections:jar:3.2.2:compile
[INFO] +- com.google.truth:truth:jar:0.29:test
[INFO] |  \- junit:junit:jar:4.13.2:test
[INFO] |     \- org.hamcrest:hamcrest-core:jar:1.3:test
[INFO] +- com.rubiconproject.oss:jchronic:jar:0.2.6:compile
[INFO] +- com.wavefront:wavefront-sdk-java:jar:3.0.0:compile
[INFO] |  \- com.google.code.findbugs:jsr305:jar:3.0.2:compile
[INFO] +- com.wavefront:wavefront-internal-reporter-java:jar:1.7.10:compile
[INFO] |  +- com.fasterxml.jackson.core:jackson-annotations:jar:2.12.3:compile
[INFO] |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-yaml:jar:2.12.3:compile
[INFO] |     \- org.yaml:snakeyaml:jar:1.27:compile
[INFO] +- com.amazonaws:aws-java-sdk-sqs:jar:1.11.946:compile
[INFO] |  +- com.amazonaws:aws-java-sdk-core:jar:1.11.946:compile
[INFO] |  |  +- software.amazon.ion:ion-java:jar:1.0.2:compile
[INFO] |  |  \- com.fasterxml.jackson.dataformat:jackson-dataformat-cbor:jar:2.12.3:compile
[INFO] |  \- com.amazonaws:jmespath-java:jar:1.11.946:compile
[INFO] +- com.beust:jcommander:jar:1.81:compile
[INFO] +- com.squareup.tape2:tape:jar:2.0.0-beta1:compile
[INFO] +- io.jaegertracing:jaeger-thrift:jar:1.2.0:compile
[INFO] |  +- io.jaegertracing:jaeger-core:jar:1.6.0:compile
[INFO] |  |  \- io.opentracing:opentracing-util:jar:0.33.0:compile
[INFO] |  |     \- io.opentracing:opentracing-noop:jar:0.33.0:compile
[INFO] |  +- org.apache.thrift:libthrift:jar:0.14.1:compile
[INFO] |  |  +- org.apache.tomcat.embed:tomcat-embed-core:jar:8.5.72:compile
[INFO] |  |  |  \- org.apache.tomcat:tomcat-annotations-api:jar:8.5.72:compile
[INFO] |  |  \- javax.annotation:javax.annotation-api:jar:1.3.2:compile
[INFO] |  \- com.squareup.okhttp3:okhttp:jar:4.2.2:compile
[INFO] |     +- com.squareup.okio:okio:jar:2.2.2:compile
[INFO] |     \- org.jetbrains.kotlin:kotlin-stdlib:jar:1.3.72:compile
[INFO] |        \- org.jetbrains.kotlin:kotlin-stdlib-common:jar:1.3.72:compile
[INFO] +- io.opentelemetry:opentelemetry-exporters-jaeger:jar:0.4.1:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk:jar:0.4.1:compile
[INFO] |  |  \- io.opentelemetry:opentelemetry-api:jar:0.4.1:compile
[INFO] |  |     \- io.opentelemetry:opentelemetry-context-prop:jar:0.4.1:compile
[INFO] |  +- io.opentelemetry:opentelemetry-sdk-contrib-otproto:jar:0.4.1:runtime
[INFO] |  |  \- io.opentelemetry:opentelemetry-proto:jar:0.4.1:runtime
[INFO] |  +- io.grpc:grpc-api:jar:1.41.0:compile
[INFO] |  |  \- io.grpc:grpc-context:jar:1.41.0:compile
[INFO] |  +- io.grpc:grpc-protobuf:jar:1.38.0:compile
[INFO] |  |  +- com.google.api.grpc:proto-google-common-protos:jar:2.0.1:compile
[INFO] |  |  \- io.grpc:grpc-protobuf-lite:jar:1.38.0:compile
[INFO] |  \- io.grpc:grpc-stub:jar:1.38.0:compile
[INFO] +- com.uber.tchannel:tchannel-core:jar:0.8.29:compile
[INFO] |  +- com.google.code.gson:gson:jar:2.8.8:compile
[INFO] |  +- io.opentracing:opentracing-api:jar:0.33.0:compile
[INFO] |  +- io.netty:netty-buffer:jar:4.1.70.Final:compile
[INFO] |  +- io.netty:netty-transport:jar:4.1.70.Final:compile
[INFO] |  +- io.netty:netty-transport-native-epoll:jar:4.1.17.Final:compile
[INFO] |  \- io.netty:netty-codec-http:jar:4.1.70.Final:compile
[INFO] +- io.zipkin.zipkin2:zipkin:jar:2.11.12:compile
[INFO] +- org.jboss.resteasy:resteasy-client:jar:3.15.2.Final:compile
[INFO] |  +- org.jboss.spec.javax.ws.rs:jboss-jaxrs-api_2.1_spec:jar:2.0.1.Final:compile
[INFO] |  +- org.jboss.logging:jboss-logging:jar:3.4.1.Final:compile
[INFO] |  \- commons-io:commons-io:jar:2.9.0:compile
[INFO] +- org.jboss.resteasy:resteasy-jackson2-provider:jar:3.15.2.Final:compile
[INFO] |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-json-provider:jar:2.11.3:compile
[INFO] |  |  +- com.fasterxml.jackson.jaxrs:jackson-jaxrs-base:jar:2.11.3:compile
[INFO] |  |  \- com.fasterxml.jackson.module:jackson-module-jaxb-annotations:jar:2.11.3:compile
[INFO] |  |     \- jakarta.activation:jakarta.activation-api:jar:1.2.1:compile
[INFO] |  \- com.github.fge:json-patch:jar:1.9:compile
[INFO] |     \- com.github.fge:jackson-coreutils:jar:1.6:compile
[INFO] |        \- com.github.fge:msg-simple:jar:1.1:compile
[INFO] |           \- com.github.fge:btf:jar:1.2:compile
[INFO] +- com.tdunning:t-digest:jar:3.2:compile
[INFO] +- org.easymock:easymock:jar:4.3:test
[INFO] |  \- org.objenesis:objenesis:jar:3.2:test
[INFO] +- org.hamcrest:hamcrest-all:jar:1.3:compile
[INFO] +- io.netty:netty-codec:jar:4.1.69.Final:compile
[INFO] |  \- io.netty:netty-common:jar:4.1.69.Final:compile
[INFO] +- io.netty:netty-resolver:jar:4.1.70.Final:compile
[INFO] +- commons-daemon:commons-daemon:jar:1.0.15:compile
[INFO] +- net.jpountz.lz4:lz4:jar:1.3.0:compile
[INFO] +- com.lmax:disruptor:jar:3.3.7:compile
[INFO] +- io.thekraken:grok:jar:0.1.4:compile
[INFO] |  +- com.github.tony19:named-regexp:jar:0.2.3:compile
[INFO] |  \- org.apache.commons:commons-lang3:jar:3.1:compile
[INFO] +- net.jafama:jafama:jar:2.1.0:compile
[INFO] +- io.grpc:grpc-netty:jar:1.38.0:compile
[INFO] |  +- io.grpc:grpc-core:jar:1.38.0:compile (version selected from constraint [1.38.0,1.38.0])
[INFO] |  |  \- com.google.android:annotations:jar:4.1.1.4:runtime
[INFO] |  +- io.netty:netty-codec-http2:jar:4.1.70.Final:compile
[INFO] |  +- io.netty:netty-handler-proxy:jar:4.1.70.Final:runtime
[INFO] |  |  \- io.netty:netty-codec-socks:jar:4.1.70.Final:runtime
[INFO] |  +- io.perfmark:perfmark-api:jar:0.23.0:runtime
[INFO] |  \- org.codehaus.mojo:animal-sniffer-annotations:jar:1.19:runtime
[INFO] +- com.google.protobuf:protobuf-java:jar:3.18.1:compile
[INFO] +- com.google.protobuf:protobuf-java-util:jar:3.18.1:compile
[INFO] +- net.openhft:chronicle-map:jar:3.20.84:compile
[INFO] |  +- com.intellij:annotations:jar:12.0:compile
[INFO] |  +- net.openhft:chronicle-core:jar:2.20.122:compile
[INFO] |  |  +- net.openhft:chronicle-analytics:jar:2.20.2:compile
[INFO] |  |  \- org.jetbrains:annotations:jar:19.0.0:compile
[INFO] |  +- net.openhft:chronicle-values:jar:2.20.80:compile
[INFO] |  |  +- net.openhft:chronicle-bytes:jar:2.20.80:compile
[INFO] |  |  \- com.squareup:javapoet:jar:1.12.1:compile
[INFO] |  +- net.openhft:chronicle-threads:jar:2.20.104:compile
[INFO] |  |  \- net.openhft:affinity:jar:3.20.0:compile
[INFO] |  +- net.openhft:chronicle-wire:jar:2.20.111:compile
[INFO] |  |  \- net.openhft:compiler:jar:2.4.0:compile
[INFO] |  +- net.openhft:chronicle-algorithms:jar:2.20.80:compile
[INFO] |  |  \- backport-util-concurrent:backport-util-concurrent:jar:3.1:compile
[INFO] |  +- net.java.dev.jna:jna:jar:5.5.0:compile
[INFO] |  +- net.java.dev.jna:jna-platform:jar:5.5.0:compile
[INFO] |  +- com.thoughtworks.xstream:xstream:jar:1.4.18:compile
[INFO] |  |  \- io.github.x-stream:mxparser:jar:1.2.2:compile
[INFO] |  |     \- xmlpull:xmlpull:jar:1.1.3.1:compile
[INFO] |  +- org.codehaus.jettison:jettison:jar:1.4.1:compile
[INFO] |  \- org.ops4j.pax.url:pax-url-aether:jar:2.6.2:compile
[INFO] |     +- org.ops4j.pax.url:pax-url-aether-support:jar:2.6.2:compile
[INFO] |     |  \- org.apache.maven.resolver:maven-resolver-impl:jar:1.3.1:compile
[INFO] |     |     +- org.apache.maven.resolver:maven-resolver-api:jar:1.3.1:compile
[INFO] |     |     +- org.apache.maven.resolver:maven-resolver-spi:jar:1.3.1:compile
[INFO] |     |     \- org.apache.maven.resolver:maven-resolver-util:jar:1.3.1:compile
[INFO] |     \- org.slf4j:jcl-over-slf4j:jar:1.6.6:compile
[INFO] +- org.slf4j:slf4j-nop:jar:1.8.0-beta4:compile
[INFO] \- org.springframework.boot:spring-boot-starter-log4j2:jar:2.5.5:compile
[INFO]    +- org.apache.logging.log4j:log4j-slf4j-impl:jar:2.14.1:compile
[INFO]    +- org.apache.logging.log4j:log4j-core:jar:2.14.1:compile
[INFO]    +- org.apache.logging.log4j:log4j-jul:jar:2.14.1:compile
[INFO]    \- org.slf4j:jul-to-slf4j:jar:1.7.32:compile
```

File: proxy/src/main/java/com/wavefront/agent/AbstractAgent.java (hunk 3 above)
Comment (laullon) [reply proxy/src/main/java/com/wavefront/agent/AbstractAgent.java]: In fact, I'm going to remove it, don't want to lose an error message.

**Review by joannak-vmware — APPROVED** [review]
LGTM

**Pull request closed without merging**